hbase-server/src/main/java/org/apache/hadoop/hbase/master/CatalogJanitor.java

```diff
@@ -173,6 +175,8 @@ int scan() throws IOException {
       this.lastReport = scanForReport();
       if (!this.lastReport.isEmpty()) {
         LOG.warn(this.lastReport.toString());
+        // expires unknown servers
+        repairUnknownServers();
```

Comment:
I guess not an issue for master (which doesn't have the separate namespace table), but elsewhere do we still have `hbase.master.namespace.init.timeout` setting an upper-bound on how long we wait for hbase:namespace to get assigned? Thinking that, waiting for CatalogJanitor to run, will be a pretty "slow" solution (up to 5min wait), and we may have a master crash if the ns init timeout is 5 mins as well as the catalog janitor's interval.